### PR TITLE
User authentication

### PR DESCRIPTION
**Pull request by Cyril-Jay**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 class ApplicationController < ActionController::Base
   before_filter :configure_permitted_parameters, if: :devise_controller?
+  before_action :authenticate_user!, except: [:index, :show]
```
Comment (jraenorris) [inline app/controllers/application_controller.rb]: nice

File: app/controllers/reviews_controller.rb [hunk 2]
```diff
@@ -2,17 +2,19 @@ class ReviewsController < ApplicationController
...
   end
 
   def create
+
     @review = Review.create(review_params)
     @review.location_id = params[:location_id]
     @location = @review.location
-
+    @review.user = current_user
+
```
Comment (jraenorris) [inline app/controllers/reviews_controller.rb]: what hound said

File: config/routes.rb [hunk 3]
```diff
@@ -3,10 +3,11 @@
   root "locations#index"
 
   resources :locations do
-    resources :reviews, only: [:index, :create, :new]
+    resources :reviews, only: [:index, :create, :new, :edit, :update]
```
Comment (jraenorris) [inline config/routes.rb]: Not sure that this should be `only: ...` anymore. Consider changing to `except: [:show]`

File: config/routes.rb [hunk 4]
```diff
@@ -3,10 +3,11 @@
...
   end
 
   devise_for :users
 
   resources :users
+
```
Comment (jraenorris) [inline config/routes.rb]: hound

File: db/migrate/20161006175931_add_user_to_reviews.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class AddUserToReviews < ActiveRecord::Migration[5.0]
```
Comment (jraenorris) [inline db/migrate/20161006175931_add_user_to_reviews.rb:2]: update your yml so these are ignored or make a comment here if its being addressed on another branch

File: spec/features/user_add_review_spec.rb [hunk 6]
```diff
@@ -1,36 +1,49 @@
...
-  scenario "adds a review for the location successfully" do
-    dumpling_king = Location.create(name_of_location: 'Dumpling King', description: "Great place for a cheap date, but not a first date")
+feature 'authenticated users can add reviews for the location' do
+  let!(:user) { FactoryGirl.create(:user) }
+  let!(:location) { FactoryGirl.create(:location) }
 
-    visit location_path(dumpling_king)
+  scenario 'an authenticated user can navigate to the review form' do
+    login_as(user)
+    # dumpling_king = Location.create(name_of_location: 'Dumpling King', description: 'Great place for a cheap date, but not a first date')
```
Comment (jraenorris) [inline spec/features/user_add_review_spec.rb]: take out commented out code. it's not doing anything besides potentially confusing you and your team

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (smkopp92) [inline app/controllers/application_controller.rb:4]: Yes! before_action is a really awesome helper function to use 👍

**Comment by smkopp92** [comment]
selfie-0
[_GitHub Selfies_](https://github.com/thieman/github-selfies/)

**Pull request merged**